pkg/resources/tag_acceptance_test.go

```diff
@@ -38,52 +57,341 @@ func TestAcc_Tag(t *testing.T) {
-					resource.TestCheckResourceAttr(resourceName, "name", id.Name()),
-					resource.TestCheckResourceAttr(resourceName, "fully_qualified_name", id.FullyQualifiedName()),
-					resource.TestCheckResourceAttr(resourceName, "database", acc.TestDatabaseName),
-					resource.TestCheckResourceAttr(resourceName, "schema", acc.TestSchemaName),
-					resource.TestCheckResourceAttr(resourceName, "allowed_values.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_values.0", ""),
-					resource.TestCheckResourceAttr(resourceName, "comment", "Terraform acceptance test"),
+				Config: config.FromModel(t, baseModel),
+				Check: assert.AssertThat(t, resourceassert.TagResource(t, baseModel.ResourceReference()).
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()).
+					HasCommentString("").
+					HasNoMaskingPolicy(),
+					assert.Check(resource.TestCheckResourceAttr(baseModel.ResourceReference(), "allowed_values.#", "0")),
+					resourceshowoutputassert.TagShowOutput(t, baseModel.ResourceReference()).
+						HasCreatedOnNotEmpty().
+						HasName(id.Name()).
+						HasDatabaseName(id.DatabaseName()).
+						HasSchemaName(id.SchemaName()).
+						HasComment("").
+						HasOwner(snowflakeroles.Accountadmin.Name()).
+						HasOwnerRoleType("ROLE"),
+					assert.Check(resource.TestCheckResourceAttr(baseModel.ResourceReference(), "show_output.0.allowed_values.#", "0")),
 				),
 			},
-
-			// test - change allowed values
+			// import without optionals
+			{
+				Config:            config.FromModel(t, baseModel),
+				ResourceName:      baseModel.ResourceReference(),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// set all fields
 			{
-				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Tag/basic"),
-				ConfigVariables: variableSet2,
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(resourceName, "name", id.Name()),
-					resource.TestCheckResourceAttr(resourceName, "fully_qualified_name", id.FullyQualifiedName()),
-					resource.TestCheckResourceAttr(resourceName, "allowed_values.#", "2"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_values.0", "alv1"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_values.1", "alv2"),
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Tag/complete"),
+				ConfigVariables: config.ConfigVariablesFromModel(t, modelWithExtraFields),
+				Check: assert.AssertThat(t, resourceassert.TagResource(t, modelWithExtraFields.ResourceReference()).
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()).
+					HasCommentString("foo"),
+					assert.Check(resource.TestCheckResourceAttr(modelWithExtraFields.ResourceReference(), "masking_policies.#", "1")),
```

Comment:
Because these fields are sets.

Reply:
Does it matter? You can have a function `HasMaskingPolicy("one", "two", "tree)` that would check for len == 3 and would check for items with `.*` like with your current approach, but it would encapsulate it in this function.

Reply:
Yes, because for sets, we have to use `TestCheckTypeSetElemAttr`, which is incompatible with `ToTerraformTestCheckFunc`. This was already discussed during the implementation of one of the secret resources.